alica_engine/include/engine/blackboard/KeyMapping.h

```diff
@@ -11,12 +12,13 @@ class Blackboard;
 class KeyMapping
 {
 public:
-    using KeyMappingList = std::vector<std::pair<std::string, std::string>>;
+    // contains tuple of parentKey, childKey and value
+    using KeyMappingList = std::vector<std::tuple<std::string, std::string, std::string>>;
```

Comment:
instead of a tuple would prefer to have a struct which conveys the intention clearly i.e.
```
struct Mapping {
    std::variant<std::string, std::string> parentKeyOrConstant;
    std::string childKey;
};
```

Reply:
Are you sure with std::variant here? It seems like this should be std::pair instead, the first value of the pair is the parentKey, the second value is the constant.

Reply:
child key can be mapped to parent key or constant, but not both, therefore I went for std::variant